utils/env_preferences/src/parse/posix.rs

```diff
@@ -0,0 +1,621 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_locale::extensions::unicode::{key, value};
+use icu_locale::extensions::Extensions;
+use icu_locale::subtags::{language, script, variant, Language, Region, Variants};
+use icu_locale::{LanguageIdentifier, Locale};
+
+// An incomplete list of POSIX language names that need to be converted to a BCP-47 equivalent
+// Some of these are approximate and not exact (e.g. "C"->"und")
+// This is based on real-world data only, so will hopefully cover most of the more common edge cases
+// Ideally, this would be part of the CLDR dataset, see discussion for more details:
```

Comment:
`opinion`: I disagree that this should be in CLDR. CLDR is a Unicode database of locale information, not a POSIX compatibility maintenance db. This file, in the crate you're building may be the best source of such mapping.

Reply:
This is a great point, especially since the `intl/locale.alias` file linked above explicitly mentions these are deprecated aliases and probably won't change. I was looking for a canonical list of _all_ language endonyms and exonyms but it seems like only the data from `intl/locale.alias` is actually needed.